src/openapi/streaming/subscription.js

```diff
@@ -205,9 +208,16 @@ function onSubscribeSuccess(referenceId, result) {
     }
     this.updatesBeforeSubscribed = null;
 
+    this.retryCounter = 0;
+
     onReadyToPerformNextAction.call(this);
 }
 
+function retrySubscription() {
+    this.onUnsubscribe();

```

Comment:
If Gateway timetous on subscribe request. Why do we need to unsubscribe? 
Also, what will happend if onUnsubscribeError returns timeout as well?